dirsrvtests/tests/suites/password/password_TPR_policy_test.py

```diff
@@ -0,0 +1,508 @@
+        ]:
+
+        user = _create_user(topo, uid, cn, sn, givenname, userpassword, gid, ou)
+    instance = (f'ou=People,{DEFAULT_SUFFIX}')
+
+    def fin():
+
+        for user1 in UserAccounts(topo.standalone, DEFAULT_SUFFIX, rdn=None).list():
+            user1.delete()
+
+    request.addfinalizer(fin)
+
+def change_pwp_parameter(topo, pwp, operation, to_do):
+    """
+    Will change password policy parameter
+    """
+    pwp1 = PwPolicyManager(topo.standalone)
+    user = pwp1.get_pwpolicy_entry(f'{pwp},{DEFAULT_SUFFIX}')
+    user.replace(operation, to_do)
+
+
+@pytest.fixture(scope="function")
+def set_global_TPR_policies(request, topo):
+    """Sets the required global password policy attributes under
+    cn=config entry
+    """
+
+    attrs = {'passwordMustChange': '',
+             'passwordTPRMaxUse': '',
+             'passwordTPRDelayExpireAt': '',
+             'passwordTPRDelayValidFrom': '',
+            }
+
+    log.info('Get the default values')
+    entry = topo.standalone.getEntry(DN_CONFIG, ldap.SCOPE_BASE,
+                                            '(objectClass=*)', attrs.keys())
+
+    for key in attrs.keys():
+        attrs[key] = entry.getValue(key)
+
+    log.info('Set the Global password policy passwordMustChange on, passwordTPRMaxUse 3')
+    log.info('passwordTPRDelayExpireAt 600, passwordTPRDelayValidFrom 6')
+    topo.standalone.config.replace_many(('passwordMustChange', 'on'),
+                                        ('passwordTPRMaxUse', '3'),
+                                        ('passwordTPRDelayExpireAt', '600'),
+                                        ('passwordTPRDelayValidFrom','6'))
+ 
+    def fin():
+        """Resets the defaults"""
+
+        log.info('Reset the defaults')
+        topo.standalone.simple_bind_s(DN_DM, PASSWORD)
+        for key in attrs.keys():
+            topo.standalone.config.replace(key, attrs[key])
+
+    request.addfinalizer(fin)
+    # A short sleep is required after the modifying password policy or cn=config
+    time.sleep(0.5)
+
+def _create_local_pwp(topo, instance):
+    """
+    For a subtree entry create a local policy
+    """
+
+    policy_props = {}
+    pwp = PwPolicyManager(topo.standalone)
+    pwadm_locpol = pwp.create_subtree_policy(instance, policy_props)
+    for attribute, value in [
+        ('pwdmustchange', 'on'),
+        ('passwordTPRMaxUse', '3'),
+        ('passwordTPRDelayExpireAt', '1800'),
+        ('passwordTPRDelayValidFrom', '5'),
+    ]:
+        pwadm_locpol.add(attribute, value)
+    log.info('Creating local policies for subtree {}'.format(instance))
+    return pwadm_locpol
+
+def get_repl_entries(topo, entry_name, attr_list):
+    """Get a list of test entries from all suppliers"""
+
+    entries_list = []
+
+    time.sleep(10)
+
+    for inst in topo.all_insts.values():
+        entries = inst.search_s(DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, "uid={}".format(entry_name), attr_list)
+        entries_list += entries
```

Comment:
We are trying not to use the direct python-ldap "_s" functions anymore (including simple_bind_s).  You can probably use the UserAccounts class here.

Reply:
Thank you! yes, i will fix this